### PR TITLE
Button: Use aria-checked for checkbox and menuitemcheckbox roles

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/11627)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -207,7 +210,9 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       'aria-describedby': ariaDescribedBy,
       'aria-disabled': isPrimaryButtonDisabled,
       'data-is-focusable': dataIsFocusable,
-      'aria-pressed': toggle ? !!checked : undefined // aria-pressed attribute should only be present for toggle buttons
+      // aria-pressed attribute should only be present for toggle buttons
+      // aria-checked attribute should only be present for toggle buttons with checkbox type role
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: I was very confused at this PR until I read this comment, this is why inline documentation is helpful 😃

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 2]
```diff
@@ -194,6 +194,10 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       (this.props as any)['data-is-focusable'] === false || (disabled && !allowDisabledFocus) || this._isSplitButton ? false : true;
 
     const isCheckboxTypeRole = role === 'menuitemcheckbox' || role === 'checkbox';
+    // if isCheckboxTypeRole, always return a checked value, otherwise only return checked value if toggle is set to true
+    // This is because role="checkbox" always needs to have an aria-checked value
+    // but our checked prop only sets aria-pressed if we mark the button as a toggle="true"
+    const checkedOrPressedValue = isCheckboxTypeRole ? !!checked : toggle === true ? !!checked : undefined;
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Do we really need the `toggle === true` check? Does `toggle` alone not work?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: I did that to make it obvious it was a condition and not a value. sometimes those nested ternaries can be annoying to read.

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]:
```jsx
const checkedOrPressedValue = isCheckboxTypeRole ? !!checked : toggle ? !!checked : undefined;
```
if isCheckboxTypeRole is false then return toggle? no..., first check for toggle===true then return !!checked or undefined

**Pull request merged**